__tests__/app.test.js

```diff
@@ -13,6 +13,22 @@ describe('GET /not-a-route', () => {
   });
 });
 
+describe('GET /api', () => {
+  it('200: responds with an object detailing all of the available endpoints', () => {
+    return request(app)
+      .get('/api')
+      .expect(200)
+      .then(({ body: { endpoints } }) => {
+        expect(endpoints).toHaveProperty('GET /api', expect.any(Object));
+        expect(endpoints).toHaveProperty('GET /api/topics', expect.any(Object));
+        expect(endpoints).toHaveProperty(
+          'GET /api/articles',
+          expect.any(Object)
+        );
+      });
```

Comment:
Good testing, but we'd need to expand this with every new endpoint added, or it's only be partially tested. To get around this, we can instead require in the endpoints.json directly, and make sure that the response from the server is the same